apps/vocabulary/main.go

```diff
@@ -0,0 +1,173 @@
+// Copyright 2020 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/golang/protobuf/proto"
+
+	pb "github.com/googleapis/gnostic/openapiv2"
+)
+
+func readDocumentFromFileWithName(filename string) (*pb.Document, error) {
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	document := &pb.Document{}
+	err = proto.Unmarshal(data, document)
+	if err != nil {
+		return nil, err
+	}
+	return document, nil
+}
+func addToMap(word string, mapName map[string]int) {
+	mapName[word] += 1
+}
+
+func processDocument(document *pb.Document, schemas map[string]int, operationId map[string]int, names map[string]int, properties map[string]int) {
+	if document.Definitions != nil {
+		for _, pair := range document.Definitions.AdditionalProperties {
+			addToMap(pair.Name, schemas)
+			processSchema(pair.Value, properties)
+		}
+	}
+	for _, pair := range document.Paths.Path {
+		v := pair.Value
+		if v.Get != nil {
```

Comment:
consider using a switch statement

Reply:
These conditions aren't mutually exclusive - a path can have any/all of Get/Post/Put/Delete defined - so since switch statements don't fallthrough by default, I have a weak preference for this series of if statements.